src/scripts/public/dom-loader.js

```diff
@@ -53,17 +57,29 @@ const bcgovBlockThemeDomLoader = () => {
                 /*
                  * Set the scroll padding to the height of the fixed header.
                  */
-                document.documentElement.style.setProperty(
-                    '--scroll-padding',
-                    headerGroup.clientHeight + 'px'
-                );
+                if ( headerGroup ) {
+                    document.documentElement.style.setProperty(
+                        '--scroll-padding',
+                        headerGroup.clientHeight + 'px'
+                    );
+                }
             }
             /**
              * Make sure header and content elements exist, create min-height calc expression for CSS.
              */
+            let footerHeight = '0px';
+            if ( footer ) {
+                footerHeight = window
+                    .getComputedStyle( footer )
+                    .getPropertyValue( 'height' );
+            }
             if ( postSingleHeaderGroup && postSingleContent ) {
                 postSingleContent.style.minHeight =
-                    'calc(100dvh - ' + headerGroupHeight + ')';
+                    'calc(100dvh - (' +
+                    headerGroupHeight +
+                    ' + ' +
+                    footerHeight +
+                    '))';
```

Comment:
It looks like you put more thought into this logic than I did. I read through it and it seems to be well structured.